pyiron_continuum/damask/damaskjob.py

```diff
@@ -17,13 +19,53 @@ def __init__(self, project, job_name):
         self._loading = None
         self._geometry = None
         self._damask_results = None
+        self._damask_cpus=2
         self.input.create_group('geometry')
         self.input.create_group('material')
         self.input.create_group('loading')
         self.output.create_group('stress')
         self.output.create_group('strain')
-        self._executable_activate()
+        #self._executable_activate()
+
+        # set up some default settings
+        self.input.geometry['grains']=4
+        self.input.geometry['size']=1.0e-5 
+        self.input.geometry['grids']=16
+        self.input.geometry['vtifile']=''
+
+        # set up default settings for load
+        self.input.loading['loadfile']=''
+        self.input.loading['time']=0.0
+        self.input.loading['step']=1
+        self.input.loading['interval']=1
+
         
+    def generatematerialyaml(self,materialphase='Aluminum',homogenizationmethod='SX',c11=106.75e+9,c12=60.41e+9,c44=28.34e+9):
+        r=damask.Rotation.from_random(self.input.geometry['grains'])
+        ### config the material yaml
+        config=damask.ConfigMaterial()
+        config=config.material_add(O=r,phase=materialphase,homogenization=homogenizationmethod)
+        config.save()
+        # for homogenization
+        str='homogenization:\n  %s:\n    N_constituents: 1\n    mechanical: {type: pass}\n'%(homogenizationmethod)
+        fin = open("material.yaml", "rt")
+        data = fin.read()
+        data = data.replace('homogenization: {}',str)
+        # for phase
+        str='phase:\n  %s:\n    lattice: cF\n    mechanical:\n      output: [F, P, F_e, F_p, L_p, O]\n      elastic: {type: Hooke, C_11: %14.5e, C_12: %14.5e, C_44:%14.5e }\n      plastic:\n        type: phenopowerlaw\n        N_sl: [12]\n        a_sl: 2.25\n        atol_xi: 1.0\n        dot_gamma_0_sl: 0.001\n        h_0_sl-sl: 75.e+6\n        h_sl-sl: [1, 1, 1.4, 1.4, 1.4, 1.4, 1.4]\n        n_sl: 20\n        output: [xi_sl]\n        xi_0_sl: [31.e+6]\n        xi_inf_sl: [63.e+6]\n'%(materialphase,c11,c12,c44)
+        data = data.replace('phase: {}',str)
+        fin.close()
+        fin = open("material.yaml", "wt")
+        fin.write(data)
+        fin.close()
+
+        self._material=self.input.material.read('material.yaml')
+    
+
+    def Run(self):
+        str="DAMASK_grid --geom %s --load %s"%(self.input.geometry['vtifile'],self.input.loading['loadfile'])
+        subprocess.run(str,shell=True)
```

Comment:
You do not need to define `Run` function and use `subprocess`. If you define the `self.executable` correctly once, the parent class includes a `run_static` methid which runs the executable.